### PR TITLE
Integration spec n+1

### PR DESCRIPTION
**Pull request by Mujeeb4582**
# Task completed

- [x] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- [x] Use Capybara to write integration tests for each view in your project.
     - User index page.
     - User show page.
     - User post index page.
     - Post-show page.

### :diamonds: Before:

1. Screenshot from 2023-03-06 17-50-22
2. Screenshot from 2023-03-06 17-53-46

### :diamonds: After: 

1. Screenshot from 2023-03-06 17-51-28

2. Screenshot from 2023-03-06 17-54-39

### CODE REVIEW

File: spec/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    it "can see the user's username." do
+      visit user_posts_path(user1)
+      expect(page).to have_content(user1.name)
+    end
+    it 'can see the number of posts the user has written.' do
+      visit user_posts_path(user1)
+      expect(page).to have_content(user1.post_counter)
+    end
+    it "can see a post's title." do
+      visit user_posts_path(user1)
```
Comment (tresorsawasawa) [inline spec/integration/post_index_spec.rb:20]: - [x] Kindly ensure that all your tests are passing as right now the `Test that asserts you can see the post's title` is failing. Make sure you fix that.
Screenshot from 2023-03-08 08-03-30

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Mujeeb4582 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post_index_spec.rb (hunk 1 above)
Comment (Mujeeb4582) [reply spec/integration/post_index_spec.rb:20]: @tresorsawasawa I check it in my side it working please look at the below image

image

Comment (Mujeeb4582) [reply spec/integration/post_index_spec.rb:20]: i don't know why it's failing at your side

Comment (Mujeeb4582) [reply spec/integration/post_index_spec.rb:20]: Please check out it's working fine my side every integration test

image

**Review by TadesseDev — APPROVED** [review]
### Hi @Mujeeb4582 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**